kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/builders.kt

```diff
@@ -6,62 +6,313 @@ import io.kotest.property.RandomSource
+   object : Arb<A>() {
+      override fun edgecase(rs: RandomSource): A? = if (edgecases.isEmpty()) null else edgecases.random(rs.random)
+      override fun sample(rs: RandomSource): Sample<A> = delegate.sample(rs)
+
+      private val delegate = arbitraryBuilder { rs -> fn(rs) }
+   }
 
 /**
  * Creates a new [Arb] that performs shrinking using the supplied [Shrinker], uses the given edge cases and
  * generates values from the given function.
  */
-fun <A> arbitrary(edgecases: List<A>, shrinker: Shrinker<A>, fn: (RandomSource) -> A): Arb<A> = object : Arb<A>() {
+fun <A> arbitrary(
+   edgecases: List<A>,
+   shrinker: Shrinker<A>,
+   fn: suspend ArbitraryBuilderSyntax.(RandomSource) -> A
+): Arb<A> = object : Arb<A>() {
    override fun edgecase(rs: RandomSource): A? = if (edgecases.isEmpty()) null else edgecases.random(rs.random)
-   override fun sample(rs: RandomSource): Sample<A> = sampleOf(fn(rs), shrinker)
+   override fun sample(rs: RandomSource): Sample<A> = delegate.sample(rs)
+
+   private val delegate = arbitraryBuilder(shrinker) { rs -> fn(rs) }
 }
 
 /**
  * Creates a new [Arb] that generates edge cases from the given [edgecaseFn] function
  * and generates samples from the given [sampleFn] function.
  */
-fun <A> arbitrary(edgecaseFn: (RandomSource) -> A?, sampleFn: (RandomSource) -> A): Arb<A> =
+fun <A> arbitrary(
+   edgecaseFn: (RandomSource) -> A?,
+   sampleFn: suspend ArbitraryBuilderSyntax.(RandomSource) -> A
+): Arb<A> =
    object : Arb<A>() {
       override fun edgecase(rs: RandomSource): A? = edgecaseFn(rs)
-      override fun sample(rs: RandomSource): Sample<A> = Sample(sampleFn(rs))
+      override fun sample(rs: RandomSource): Sample<A> = delegate.sample(rs)
+
+      private val delegate: Arb<A> = arbitraryBuilder { rs -> sampleFn(rs) }
    }
 
 /**
  * Creates a new [Arb] that generates edge cases from the given [edgecaseFn] function,
- * performs shrinking using the supplied [Shrinker, and generates samples from the given [sampleFn] function.
+ * performs shrinking using the supplied [Shrinker], and generates samples from the given [sampleFn] function.
  */
 fun <A> arbitrary(
    edgecaseFn: (RandomSource) -> A?,
    shrinker: Shrinker<A>,
-   sampleFn: (RandomSource) -> A
+   sampleFn: suspend ArbitraryBuilderSyntax.(RandomSource) -> A
 ): Arb<A> =
    object : Arb<A>() {
       override fun edgecase(rs: RandomSource): A? = edgecaseFn(rs)
-      override fun sample(rs: RandomSource): Sample<A> = sampleOf(sampleFn(rs), shrinker)
+      override fun sample(rs: RandomSource): Sample<A> = delegate.sample(rs)
+
+      private val delegate: Arb<A> = arbitraryBuilder(shrinker) { rs -> sampleFn(rs) }
+   }
+
+@Suppress("ClassName")
+object arbitrary {
+   /**
+    * Creates a new [Arb] that performs no shrinking, has no edge cases and
+    * generates values from the given function.
+    */
+   suspend inline fun <A> suspendable(
```

Comment:
I thought we were going with generateArbitrary to match generateSequence from the std lib.
Not totally bothered but suspendable isn't clear that it's a builder at all to me.

Reply:
generateSequence is not suspend apparently. I was trying to find a better syntax and tried to brainstorm this. Looking at arrow, they have `either` and `either.eager` for instance. I'll ping you in slack for better guidance.
 
```kotlin
// non-suspend
arbitrary { ... } 

// alt 1
arbitrary.suspendable { ... }

// alt 2
suspendArbitrary { ... }

// alt 3
generateArbitrary { ... }
```

Reply:
i'm overthinking it, `generateArbitrary` is a great name 🚀